### PR TITLE
More layout elevation profile UI work

### PR DESCRIPTION
**Pull request by nyalldawson**
This next step in layout elevation profiles implements the following changes:

- Adds a toolbar to elevation profile widget with button to copy settings from a existing elevation profile dock. This is currently the only way to get the profile curve set for layout elevation profiles. (In follow ups I may also expose a "draw profile line" tool for the layout item, but the UX there will be a little awkward as it would have to switch back to the map canvas for the user to draw the curve)
- Exposes the profile tolerance distance setting, and adds data-defined support for this setting
- Adds a manual "refresh plot" button to the item properties toolbar

image

(This is the last ground-work PR for the layout items -- the next one will implement the actual rendering of the profile)

### CODE REVIEW

**Comment by nicogodet** [comment]
> In follow ups I may also expose a "draw profile line" tool for the layout item, but the UX there will be a little awkward as it would have to switch back to the map canvas for the user to draw the curve

A "synchronize" feature to always update layout elevation profile based on profile dock ?

**Comment by nyalldawson** [comment]
@nicogodet 

> A "synchronize" feature to always update layout elevation profile based on profile dock ?

That's a little tricky, given that there's potentially multiple elevation docks open at any stage (and also because these docks aren't saved/restored along with the project)

File: src/gui/layout/qgslayoutelevationprofilewidget.cpp [hunk 1]
```diff
@@ -431,6 +475,80 @@ QgsExpressionContext QgsLayoutElevationProfileWidget::createExpressionContext()
   return mProfile->createExpressionContext();
 }
 
+void QgsLayoutElevationProfileWidget::setDesignerInterface( QgsLayoutDesignerInterface *iface )
+{
+  mInterface = iface;
+  QgsLayoutItemBaseWidget::setDesignerInterface( iface );
+}
+
+void QgsLayoutElevationProfileWidget::copySettingsFromProfileCanvas( QgsElevationProfileCanvas *canvas )
```
Comment (nirvn) [inline src/gui/layout/qgslayoutelevationprofilewidget.cpp:484]: Would it be worth adding a test for this function?

Comment (nyalldawson) [reply src/gui/layout/qgslayoutelevationprofilewidget.cpp:484]: I'll defer till after freeze :wink:

**Pull request merged**